Some polishing of the random/hungry/followers notebooks and fixing stuff (collab2 ).

dimkab opened the pull request:
* RHF notebooks documentation and improvements
* some code fixes

emackev reviewed (commented):
It's looking nice!! Here are a couple areas where a little more documentation would be useful:

- Referencing the [paper](https://www.nature.com/articles/s41598-024-71931-0), both in the overall readme, and in the beginning of each notebook (e.g. in the beginning of the RHF notebooks, reference fig 2 )
- In the Derived Quantities section of the Random notebook, it'd be helpful to explain what the model looks like, possibly with an equation. Readers may be confused what a predictor/score is, and how they'll be used. It's important that they know that the model output is the next step score. (this sentence in particular needs more context/explanation: "At the inference stage, we shall fit the scores to a linear combination of predictors with unknown coefficients.")
- More explanation of Pyro models (in the Random notebook, where it says "Now we create the Pyro model.", people may not know what Pyro is. It should have a brief explanation, and link to documentation/repo)
- Access predictor should be explained 


A couple other things
- Follower birds, can the initial conditions have the agents farther apart (so it doesn't start with them already clumped together)?
- Follower birds, the animation isn't loaded in the notebook on git
- Follower birds, the histogram of the posterior distribution at the end of the notebook is missing
- Consider changing scatter plots to heatmaps

dimkab commented:
Thanks @emackev  for the feedback! I've implemented everything except density plots.
I've added the option to specify initial forager positions. However, for the followers simulation, setting it to something other than the default produces not great results. You are welcome to play with the parameter (uncomment `initial_positions` below):
```
follower_sim = rhf.add_follower_foragers(
    follower_sim,
    num_follower_foragers=3,
    visibility_range=6,
    getting_worse=0.5,
    optimal=3,
    #initial_positions = np.array([[25, 25], [30, 30], [35,35]])
)
```
